### DISCUSSION TITLE
Tree crown segmentation using yolo8

### DISCUSSION THREAD

**Discussion by arabinda90** [ultralytics/ultralytics#8272] [Q&A]
Dear community thank you to each of the members

I want to extract the tree crown boundary using the Yolo8 model. After training the model when I predict the RGB image each tree has multiple polygons but it should be a single polygon for a tree. Have you any idea how to address this issue?
![trial](https://github.com/ultralytics/ultralytics/assets/84968673/3c4a65f0-afca-4e40-91fb-82a10130c3df)

**Comment by glenn-jocher** [comment]
Hello! Thanks for reaching out with your question. 🌳

It sounds like you're experiencing an issue where individual trees are being detected with multiple polygons rather than a single one. This could be due to overlapping detections or the model not learning the tree crowns as single entities.

One approach to address this could be to adjust the Non-Maximum Suppression (NMS) threshold during prediction. This can help reduce overlapping detections. Here's a snippet on how you might adjust the NMS threshold:

```python
from ultralytics import YOLO

# Load your trained model
model = YOLO('path/to/your/trained_model.pt')

# Set a higher NMS threshold
results = model.predict('path/to/your/image.jpg', iou=0.5)  # Adjust the iou value as needed
```

If this doesn't solve the issue, you might need to look into further tuning your training process, possibly by adjusting the dataset annotations or the loss function to better capture the concept of a single tree crown.

Hope this helps! Keep us posted on your progress. 🌲
